src/Illuminate/Foundation/Configuration/ApplicationBuilder.php

```diff
@@ -159,12 +159,17 @@ public function withRouting(?Closure $using = null,
         string $apiPrefix = 'api',
         ?callable $then = null)
     {
-        if (is_null($using) && (is_string($web) || is_array($web) || is_string($api) || is_array($api) || is_string($pages) || is_string($health)) || is_callable($then)) {
-            $using = $this->buildRoutingCallback($web, $api, $pages, $health, $apiPrefix, $then);
+        if (is_null($using) && (is_string($web) || is_array($web) || is_string($api) || is_array($api) || is_string($pages)) || is_callable($then)) {
+            $using = $this->buildRoutingCallback($web, $api, $pages, $apiPrefix, $then);
+        }
 
-            if (is_string($health)) {
-                PreventRequestsDuringMaintenance::except($health);
-            }
+        if (is_string($health)) {
+            $using = function () use ($health, $using) {
+                $this->buildHealthCheckRoute($health);
+                call_user_func($using, $this->app);
```

Comment:
This line should be:

```php
$this->app->call($using);
```

To preserve the current behavior, as in:

https://github.com/laravel/framework/blob/df12a087731b37708cfbba72172a4c381363fed2/src/Illuminate/Foundation/Support/Providers/RouteServiceProvider.php#L162

Reply:
Also, `$using` in this point, can be `null`. A check for that should be added.

An app can be built without specifying no routes, just the `$commands` and `$channels` parameters.